src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray`1+Builder.cs

```diff
@@ -49,7 +49,41 @@ internal Builder()
             }
 
             /// <summary>
-            /// Gets or sets the length of the array.
+            /// Get and sets the length of the internal array.  When set the internal array is 
+            /// is reallocated to the given capacity if it is not already the specified length.
+            /// </summary>
+            public int Capacity
+            {
+                get { return _elements.Length; }
+                set
```

Comment:
Actually it wouldn't. I missed this on my first read too. Due to the argument validation, the only way `value==0` is allowed is when `_elements.Length == 0`. When that occurs `Array.Resize` is a NOP.